src/SphereXEngine.sol

```diff
@@ -185,26 +236,38 @@ contract SphereXEngine is ISphereXEngine, AccessControlDefaultAdminRules {
     function _addCfElementFunctionEntry(int256 num) internal {
         require(num > 0, "SphereX error: expected positive num");
         FlowConfiguration memory flowConfig = _flowConfig;
+        EngineConfig memory engineConfig = _engineConfig;
 
         // Upon entry to a new function we should check if we are at the same transaction
         // or a new one. in case of a new one we need to reinit the currentPattern, and save
         // the new transaction "boundry" (block.number+tx.origin+block.timestamp+block.difficulty)
-        bytes3 currentTxBoundaryHash =
-            bytes3(keccak256(abi.encode(block.number, tx.origin, block.timestamp, block.difficulty)));
-        if (currentTxBoundaryHash != flowConfig.txBoundaryHash) {
+        bytes16 currentTxBoundaryHash =
+            bytes16(keccak256(abi.encode(block.number, tx.origin, block.timestamp, block.difficulty)));
+        if (currentTxBoundaryHash != engineConfig.txBoundaryHash) {
             flowConfig.pattern = PATTERN_START;
-            flowConfig.txBoundaryHash = currentTxBoundaryHash;
+            engineConfig.txBoundaryHash = currentTxBoundaryHash;
+            flowConfig.enforce = false;
             if (flowConfig.depth != DEPTH_START) {
                 // This is an edge case we (and the client) should be able to monitor easily.
                 emit TxStartedAtIrregularDepth();
                 flowConfig.depth = DEPTH_START;
             }
         }
 
+        if (_isSelectiveTxfActivated(engineConfig.rules)) {
+            // if we are not in enformecnt mode then check if the current function switch it on
+            if (!flowConfig.enforce) {
+                if (_enforceFunction[uint256(num)]) {
+                    flowConfig.enforce = true;
+                }
+            }
+        }
+
         flowConfig.pattern = uint216(bytes27(keccak256(abi.encode(num, flowConfig.pattern))));
         ++flowConfig.depth;
 
         _flowConfig = flowConfig;
```

Comment:
Isn't it posible flowConfig won't get modified?

Reply:
no, the depth will always be changed